dbms/src/Interpreters/Join.cpp

```diff
@@ -823,147 +823,173 @@ void Join::handleOtherConditions(Block & block, std::unique_ptr<IColumn::Filter>
 
 Block Join::joinBlockHash(ProbeProcessInfo & probe_process_info) const
 {
-    Block block = probe_process_info.block;
-    size_t keys_size = key_names_left.size();
-
-    /// Rare case, when keys are constant. To avoid code bloat, simply materialize them.
-    /// Note: this variable can't be removed because it will take smart pointers' lifecycle to the end of this function.
-    Columns materialized_columns;
-    ColumnRawPtrs key_columns = extractAndMaterializeKeyColumns(block, materialized_columns, key_names_left);
-
-    /// Keys with NULL value in any column won't join to anything.
-    ColumnPtr null_map_holder;
-    ConstNullMapPtr null_map{};
-    extractNestedColumnsAndNullMap(key_columns, null_map_holder, null_map);
-    /// reuse null_map to record the filtered rows, the rows contains NULL or does not
-    /// match the join filter won't join to anything
-    recordFilteredRows(block, non_equal_conditions.left_filter_column, null_map_holder, null_map);
-
-    size_t existing_columns = block.columns();
-
-    /** If you use FULL or RIGHT JOIN, then the columns from the "left" table must be materialized.
+    std::vector<Block> result_blocks;
+    size_t result_rows = 0;
+    while (true)
+    {
+        probe_process_info.updateStartRow();
```

Comment:
Actually wrap the original `Join::joinBlockHash()` by a `while` loop, the code insde the loop is not changed.

Reply:
how about wrap the code like
```
Block doJoinBlockHash();

Block joinBlockHash()
{
    while (true)
    {
        doJoinBlockHash();
    }
}
```
?

Reply:
Done